packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/Controls/DataTable.tsx

```diff
@@ -0,0 +1,278 @@
+  }));
+
+  // Sort the data based on the selected column and direction
+  const { selection, direction } = sortState;
+  const selectedColumn = columnData.find((column) => column.key === selection);
+
+  if (selectedColumn?.compareFn) {
+    data.sort((a, b) => {
+      const comparison = selectedColumn.compareFn!(a, b);
+      return direction === 'ascending' ? comparison : -comparison;
+    });
+  }
+
+  // Generate column headers
+  const columns = columnData.map((column) =>
+    getColumnItem({
+      columnData: column,
+      selection: selectedColumn!.key,
+      direction,
+      onTableHeaderClick,
+    })
+  );
+
+  // Add the download column header
+  columns.push({
+    'aria-label': 'Download',
+    key: 'th_download',
+    className: `${TABLE_HEADER_CLASS_NAME} ${TABLE_HEADER_CLASS_NAME}--download`,
+    'aria-sort': 'none',
+  });
+
+  // Generate rows for the table
+  const rows = data.map((item, index) => {
+    const { type } = item;
+
+    return [
+      {
+        className: `${TABLE_DATA_CLASS_NAME} ${TABLE_DATA_CLASS_NAME}--key`,
+        key: `td-name-${index}`,
+        children:
+          type === 'FILE' ? (
+            <SpanElement className={TABLE_DATA_TEXT_CLASS_NAME}>
+              <IconElement className={TABLE_DATA_ICON_CLASS} variant="file" />
+              {item.key}
+            </SpanElement>
+          ) : (
+            <ButtonElement
+              className={TABLE_DATA_BUTTON_CLASS}
+              onClick={() => onLocationItemFolderClick(item.key)}
+              variant="table-data"
+            >
+              <IconElement className={TABLE_DATA_ICON_CLASS} variant="folder" />
+              {item.key}
+            </ButtonElement>
+          ),
+      },
+      {
+        className: `${TABLE_DATA_CLASS_NAME} ${TABLE_DATA_CLASS_NAME}--type`,
+        key: `td-type-${index}`,
+        children: (
+          <SpanElement className={TABLE_DATA_TEXT_CLASS_NAME}>
+            {type === 'FILE' ? item.fileExt : 'Folder'}
+          </SpanElement>
+        ),
+      },
+      {
+        className: `${TABLE_DATA_CLASS_NAME} ${TABLE_DATA_CLASS_NAME}--lastModified`,
+        key: `td-lastModified-${index}`,
+        children: (
+          <SpanElement className={TABLE_DATA_TEXT_CLASS_NAME}>
+            {type === 'FILE'
+              ? new Date(item.lastModified!).toLocaleString()
+              : ''}
+          </SpanElement>
+        ),
+      },
+      {
+        className: `${TABLE_DATA_CLASS_NAME} ${TABLE_DATA_CLASS_NAME}--size`,
+        key: `td-size-${index}`,
+        children: (
+          <SpanElement className={TABLE_DATA_TEXT_CLASS_NAME}>
+            {type === 'FILE' ? humanFileSize(item.size ?? 0, true) : ''}
+          </SpanElement>
+        ),
+      },
+      {
+        className: `${TABLE_DATA_CLASS_NAME} ${TABLE_DATA_CLASS_NAME}--download`,
+        key: `td-download-${index}`,
+        children: (
+          <SpanElement className={TABLE_DATA_TEXT_CLASS_NAME}>
```

Comment:
This SpanElement wrapping the DownloadControl can be removed (the button styles on Download already supply the padding/string truncation that plain text cells would otherwise need).

Reply:
got it, thanks Heather. Just pushed up the change to remove the span